app/controllers/api/v1/PostsController.rb

```diff
@@ -0,0 +1,95 @@
+class Api::V1::PostsController < ApplicationController
+  include Authentication
+
+  def index
+    posts = Post.all
+
+    if posts
+      render json: {data: posts}, status: :ok
+    else
+      render json: posts.errors, status: :bad_request
+    end
+  end
+
+  def create
+    if authenticate
+      user = User.find_by(username: request.headers[:username])
+      post = Post.new(title: post_params[:title], description: post_params[:description], slug: "", comments_count: 0, likes_count: 0, user: user)
+
+      if post.save
+        post.update(slug: "api/v1/posts/" << String(post.id))
+        render json: {data: post}, status: 201
+      else
+        render json: post.errors, status: :unprocessale_entity
+      end
+    else
+      render status: 401
+    end
+  end
+
+  def show
+    begin
+    post = Post.find(params[:id])
```

Comment:
find raises an exception, where as find_by does not.
No need to raise an exception in all the cases. Better use find_by and check using if else condition instead of exception handling.
The same goes for all the cases where you have used `find`

Reply:
Used `.exists?` to check it exists and continue further using if-else